[quant][graphmode][fx] Add fbgemm backend_config_dict

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #64288
* #64135

Summary:
This is just to setup the file structure and unblock experimentation.
The format for backend_config_dict will change in the future

Test Plan:
python test/test_quantization.py TestQuantizeFx
python test/test_quantization.py TestQuantizeFxOps

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D30699457](https://our.internmc.facebook.com/intern/diff/D30699457)

jerryzh168 committed:
[quant][graphmode][fx] Add fbgemm backend_config_dict

Summary:
This is just to setup the file structure and unblock experimentation.
The format for backend_config_dict will change in the future

Test Plan:
python test/test_quantization.py TestQuantizeFx
python test/test_quantization.py TestQuantizeFxOps

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

vkuzo commented on torch/quantization/fx/backend_config_dict/fbgemm.py:
```diff
@@ -0,0 +1,12 @@
+    """
+    # TODO: add output_activation_post_process_map
+    return {
+        "quant_patterns": get_default_quant_patterns()
```
might be good to clarify further that this API will definitely change, and it's 100% unsafe for customers to depend on this in any way until further notice.

vkuzo commented on torch/quantization/fx/backend_config_dict/fbgemm.py:
```diff
@@ -0,0 +1,12 @@
+
+def get_fbgemm_backend_config_dict():
+    """ Get the backend config dictionary for fbgemm backend
+    NOTE: Current api is not final, it's just to unblock experimentation for new backends
```
can we also add some checking that the values are valid, or add a TODO to do it before users start using the API?

jerryzh168 replied:
you mean `validate_backend_config_dict`? yeah we have that in __init__.py

jerryzh168 committed:
Update on "[quant][graphmode][fx] Add fbgemm backend_config_dict"


Summary:
This is just to setup the file structure and unblock experimentation.
The format for backend_config_dict will change in the future

Test Plan:
python test/test_quantization.py TestQuantizeFx
python test/test_quantization.py TestQuantizeFxOps

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 commented:
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30699457).

jerryzh168 committed:
Update on "[quant][graphmode][fx] Add fbgemm backend_config_dict"


Summary:
This is just to setup the file structure and unblock experimentation.
The format for backend_config_dict will change in the future

Test Plan:
python test/test_quantization.py TestQuantizeFx
python test/test_quantization.py TestQuantizeFxOps

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D30699457](https://our.internmc.facebook.com/intern/diff/D30699457)

[ghstack-poisoned]

jerryzh168 commented:
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30699457).
